OpenSim/Simulation/InverseKinematicsSolver.cpp

```diff
@@ -187,29 +184,29 @@ double InverseKinematicsSolver::computeCurrentSquaredMarkerError(int markerIndex
         throw Exception("InverseKinematicsSolver::computeCurrentMarkerSquaredError: invalid markerIndex.");
 }
 
-/** Compute and return the distance errors between all model marker and observations. */
+/* Compute and return the distance errors between all model marker and observations. */
 void InverseKinematicsSolver::computeCurrentSquaredMarkerErrors(SimTK::Array_<double> &markerErrors)
 {
     markerErrors.resize(_markerAssemblyCondition->getNumMarkers());
     for(unsigned int i=0; i<markerErrors.size(); i++)
         markerErrors[i] = _markerAssemblyCondition->findCurrentMarkerErrorSquared(SimTK::Markers::MarkerIx(i));
 }
 
-/** Marker errors are reported in order different from tasks file or model, find name corresponding to passed in index  */
+/* Marker errors are reported in order different from tasks file or model, find name corresponding to passed in index  */
 std::string InverseKinematicsSolver::getMarkerNameForIndex(int markerIndex) const
 {
     return _markerAssemblyCondition->getMarkerName(SimTK::Markers::MarkerIx(markerIndex));
 }
 
 
-/** Internal method to convert the MarkerReferences into additional goals of the 
+/* Internal method to convert the MarkerReferences into additional goals of the 
     of the base assembly solver, that is going to do the assembly.  */
 void InverseKinematicsSolver::setupGoals(SimTK::State &s)
 {
     // Setup coordinates performed by the base class
     AssemblySolver::setupGoals(s);
 
-    _markerAssemblyCondition = new SimTK::Markers();
+    _markerAssemblyCondition.reset(new SimTK::Markers());
```

Comment:
This type of situation has come up before ( (in Coordinate)[https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/SimbodyEngine/Coordinate.cpp#L182]), since the Assembly is actually the owner. @sherm1 came up with a way to deal with this; see the link. Can you change the code to be like the following?

```cpp
std::unique_ptr<SimTK::Markers> condOwner(new SimTK::Markers());
_markerAssemblyCondition.reset(condOnwer.get());
...
updAssembler().adoptAssemblyGoal(condOwner.release()); // condOwner gives up ownership
```